lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -825,4 +853,163 @@ class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
       }
     }
   }
+
+  Recipe? _recipeOnDevice(recipeNumber) {
+    List<Recipe> deviceRecipes = List<Recipe>.from(state.recipes);
+    for (Recipe recipe in deviceRecipes) {
+      if (recipe.recipeNumber == recipeNumber) {
+        return recipe;
+      }
+    }
+    return null;
+  }
+
+  Future<void> _onRecipeImported(
+      RecipeImported event, Emitter<RecipeState> emit) async {
+    emit(state.copyWith(recipeImportStatus: '')); // Clear status for snackbar
+    FilePickerResult? result = await FilePicker.platform.pickFiles(
+        // TODO: Implement extension restriction once csv is supported by package
+        // type: FileType.custom,
+        // allowedExtensions: ['csv'],
+        );
+
+    if (result != null) {
+      final input = File(result.files.single.path!).openRead();
+      List data = await input
+          .transform(utf8.decoder)
+          .transform(const CsvToListConverter())
+          .toList();
+      data.removeAt(0); // Remove headings line in csv
+
+      List<Recipe> newRecipes = [];
+      List<String> newerRecipesRecipeIds = [];
+      List<Recipe> existingRecipes = List<Recipe>.from(state.recipes);
+
+      for (List row in data) {
+        if (row.length < 12) {
+          emit(state.copyWith(
+              recipeImportStatus: 'Import file has too few columns'));
+          return;
+        }
+        String recipeNumber = row[0];
+        String date = row[1];
+        String recipeName = row[2];
+        String recipeType = row[3];
+        String recipeAttribute = row[4];
+        String recipeMeasurement = row[5];
+        String recipeProbeName = row[6];
+        String recipeProbeAnswers = row[7];
+        String ingredientName = row[8];
+        String ingredientDescription = row[9];
+        String ingredientCookingState = row[10];
+        String ingredientMeasurement = row[11];
+
+        Recipe? recipeOnDevice = _recipeOnDevice(recipeNumber);
+
+        if (newerRecipesRecipeIds.contains(recipeNumber)) {
+          continue;
+        } else if (recipeOnDevice != null) {
+          if (DateTime.parse(recipeOnDevice.date)
+              .isAfter(DateTime.parse(date))) {
+            newerRecipesRecipeIds.add(recipeNumber);
+            continue;
+          } else {
+            existingRecipes.remove(recipeOnDevice);
+          }
+        }
+
+        Recipe recipe = Recipe(
+            recipeNumber: recipeNumber,
+            date: date,
+            recipeName: recipeName,
+            recipeType: recipeType.trim() + ' Recipe',
+            measurements: const [],
+            saved: true);
+
+        for (Recipe existingRecipe in newRecipes) {
+          if (existingRecipe.recipeNumber == recipeNumber) {
+            recipe = existingRecipe;
+            newRecipes.remove(existingRecipe);
+            break;
+          }
+        }
```

Comment:
I see the need for such implementation using the three lists defined above (newRecipes, newerRecipesRecipeIds, existingRecipes), but I think it would be nicer to have smaller subfunctions that are well-named do this work so that the whole recipe importing method is shorter and easier to understand. Again, we can just add a todo comment for this now.

Reply:
Also, if I understand this rewrites the current recipes with matching IDs, maybe we should let the user know this is the case, or add a todo to implement some sort of user-choice for the future, where the user would be able to choose if to overwrite.

Reply:
Have added the todos.